src/runtime_src/core/pcie/driver/linux/xocl/subdev/cu_xgq.c

```diff
@@ -57,8 +57,7 @@ static void cu_xgq_check(void *core, struct xcu_status *status, bool force)
 	struct xrt_cu_xgq *cu_xgq = core;
 
 	status->num_ready = 1;
-	while (!xocl_xgq_check_response(cu_xgq->xgq, cu_xgq->xgq_client_id));
-	status->new_status = 0x4;
+	while (!xocl_xgq_check_response(cu_xgq->xgq, cu_xgq->xgq_client_id, &status->new_status));
```

Comment:
The "new_status" is a HLS style status (bit 2 is ap_idle, bit 1 is ap_done, bit 0 is ap_start etc.).
I saw xocl_xgq_check_response() is directly assign "xcmd->status" to it. This is in correct. It will make xbutil tool cannot show proper CU status.

Here you will need to convert command status to HLS CU status. "Like KDS_COMPLETE -> ap_idle"